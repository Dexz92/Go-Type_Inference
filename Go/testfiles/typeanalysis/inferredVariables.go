func main() {
  a := 1;
  b := 2;
  c := a + b;
  var d int = c;
  var e = 2.0;
  var f float64 = e + 3.5;
  var g int = d + f;

  i1 := 2;
  i2 := 3;
  b1 := true;
  var i3 int = i1 + i2;
  var i4 int = i1 + b1;
  var i5 = i3 + 5;
  var b2 bool = i5;
  var i6 int = i1 + i2 * i3 + i5;
  var f2 float64 = i1 + i2 * i3 + i5;
}
