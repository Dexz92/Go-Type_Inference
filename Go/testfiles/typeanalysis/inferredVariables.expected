Error at line 8: Invalid operation: d + f (mismatched types: int and float)
Error at line 14: Invalid operation: i1 + b1 (mismatched types: int and bool)
Error at line 16: Trying to assign int to bool variable
Error at line 18: Trying to assign int to float variable
